### ISSUE TITLE
hugo issues

### ISSUE THREAD

**Issue by mjpost** [acl-org/acl-anthology#2285]
Hugo seems to need a lot of memory, and it is killing our publish actions.

Here's a local build with `hugo v0.108.0+extended darwin/amd64 BuildDate=unknown` build on my Mac.

```
Alloc = 4.5 GB
TotalAlloc = 59.4 GB
Sys = 8.0 GB
NumGC = 110
```

And the summary printed at the end:

```
                   |   EN
-------------------+---------
  Pages            | 153949
  Paginator pages  |      0
  Non-page files   |      2
  Static files     | 253548
  Processed images |      0
  Aliases          |      0
  Sitemaps         |      1
  Cleaned          |      0

Total in 539484 ms
```

It seems like there must be a way to improve the speed and memory usage.

**Comment by akoehn** [comment]
That does at least not look too bad -- maybe we should upgrade the hugo version we use in our workflows, it is still at 0.58.

I had some code for a new hugo version somewhere and can create a PR for it -- but as the anthology grows we will hit that barrier again at some point in time :-/

**Comment by akoehn** [comment]
yes, now I remembered that the change already made it to the main branch.  Anyway, upgrading to a new version after 50(!) releases does not sound like a bad idea, so I have made a PR: https://github.com/acl-org/acl-anthology/pull/2287

**Comment by akoehn** [comment]
I think we can also remove the check-build workflow, as we also run the preview workflow, which runs a superset of the check-build wf (as far as I understand).

**Comment by mbollmann** [comment]
@mjpost Can you run and post your local memory alloc with the latest hugo version, for comparison?

**Comment by mjpost** [comment]
Here's from my local Macbook Pro:

```
                   |   EN
-------------------+---------
  Pages            | 152168
  Paginator pages  |      0
  Non-page files   |      1
  Static files     | 250745
  Processed images |      0
  Aliases          |      0
  Sitemaps         |      1
  Cleaned          |      0

Alloc = 4.3 GB
TotalAlloc = 52.7 GB
Sys = 6.9 GB
NumGC = 68

Total in 1978199 ms
```

**Comment by JohnG1985** [comment]
Hugo is a static site generator that can be used to build websites quickly and efficiently. It is designed to be fast and efficient, but like any program, it can use a lot of memory if it is not used properly.

There are a few ways you can improve the memory usage and speed of Hugo when building your website. Some possible approaches include the following:

Use the latest version of Hugo: The developers of Hugo are constantly working to improve the performance and efficiency of the program. Updating to the latest version of Hugo can help reduce memory usage and improve build times.

Use the "extended" version of Hugo: Hugo offers two versions: the standard version and the extended version. The extended version includes additional features and functionality that may be useful for your website, but it also uses more memory. If you are using the standard version of Hugo and are experiencing high memory usage, you may want to consider switching to the extended version.

Use the --minify flag when building your website: Hugo has a built-in minification tool that can be used to reduce the size of your website's HTML, CSS, and JavaScript files. This can help reduce the amount of memory that Hugo uses when building your website. To use the minification tool, you can add the --minify flag when running the hugo command to build your website.

Use the --ignoreCache flag when building your website: Hugo uses a cache to store information about your website in order to improve build times. However, if you are making a lot of changes to your website, this cache can become large and consume a lot of memory. Using the --ignoreCache flag when building your website can help reduce memory usage by forcing Hugo to rebuild the cache from scratch.

Use the --gc flag when building your website: Hugo's garbage collector is responsible for cleaning up unused objects and freeing up memory. By default, Hugo uses a conservative garbage collector that may not be as efficient as possible. Using the --gc flag when building your website can enable Hugo's more aggressive garbage collector, which can help reduce memory usage.

By using these techniques, you should be able to improve the memory usage and build times of Hugo when building your website. However, it is important to keep in mind that the exact steps you need to take will depend on your specific website and the changes you are making. Experimenting with different approaches and settings can help you find the optimal configuration for your website.

**Comment by mbollmann** [comment]
For reference, [this `MemStats` documentation](https://pkg.go.dev/runtime#MemStats) contains explanations of what the fields mean. I believe the most relevant field is either `Sys` (_"the total bytes of memory obtained from the OS"_) or the maximum value of `Alloc` over time.

From my local desktop machine (Intel Core i5-12400F):

```
hugo v0.111.3-5d4eb5154e1fed125ca8e9b5a0315c4180dab192+extended linux/amd64 BuildDate=2023-03-12T11:40:50Z VendorInfo=gohugoio

... 

Alloc = 5.1 GB
TotalAlloc = 54.6 GB
Sys = 8.0 GB
NumGC = 62


                   |   EN    
-------------------+---------
  Pages            | 158206  
  Paginator pages  |      0  
  Non-page files   |      1  
  Static files     | 260039  
  Processed images |      0  
  Aliases          |      0  
  Sitemaps         |      1  
  Cleaned          |      0  

Total in 69848 ms
```

Maximum `Alloc` during the build was `6.5 GB`.

**Comment by mjpost** [comment]
Closed with #2286.
